.github/workflows/test.yml

```diff
@@ -0,0 +1,129 @@
+---
+name: Run DTS tests
+on:
+  pull_request:
+    branches:
+      - 'main'
+jobs:
+  build-dts:
+    uses: ./.github/workflows/build.yml
+    with:
+      cacheless: false
+  run-tests:
+    name: Run DTS tests
+    if: ${{ github.head_ref == 'develop' && contains(join(needs.*.result, ','), 'success') }}
+    needs: build
+    runs-on:
+      labels: dts-builder
+    steps:
+      - name: Checkout OSFV repo
+        uses: actions/checkout@v4
+        with:
+          repository: 'Dasharo/open-source-firmware-validation'
+          path: 'open-source-firmware-validation'
+          submodules: 'recursive'
+          ref: 'develop'
+      - name: Run IPXE server
+        shell: bash
+        run: |
+          mkdir ipxe
+          cp build/tmp/deploy/images/genericx86-64/dts-base-image-genericx86-64.cpio.gz ipxe
+          cp build/tmp/deploy/images/genericx86-64/bzImage ipxe
+          echo -e "\n
+          #!ipxe\n
+          imgfetch --name file_kernel bzImage\n
+          imgfetch --name file_initrd dts-base-image-genericx86-64.cpio.gz\n
+          kernel file_kernel root=/dev/nfs initrd=file_initrd\n
+          boot" > ipxe/dts.ipxe
+          cd ipxe && python3-m http.server 4321 &
+      - name: Install requirements
+        shell: bash
+        run: |
+          cd open-source-firmware-validation
+          python3 -m virtualenv venv
+          source venv/bin/activate
+          pip install -r requirements.txt
+      - name: Run QEMU
+        shell: bash
+        id: run_qemu
+        run: |
+          cd open-source-firmware-validation/scripts/ci
+          qemu-img create -f qcow2 qemu-data/hdd.qcow 20G
+          ./qemu-run.sh nographic os &
+      - name: Create directory for logs
+        shell: bash
+        id: log_dirs
+        run: |
+          timestamp=$(date -u +%Y-%m-%dT%H:%M:%S%Z)
+          directory="/tmp/dts-test-ci-${timestamp}"
+          mkdir $directory
+          echo "directory=$directory" >> "$GITHUB_OUTPUT"
+      - name: Run tests
+        shell: bash
+        env:
+          LOG_DIR: ${{ steps.log_dirs.outputs.directory }}
+        run: |
+          cd open-source-firmware-validation
+          source venv/bin/activate
+          # This file is already present on dts-builder.
+          # It contains credentials for DPP subscriptions.
+          # It has the following form:
+          # DPP_PASSWORD="..."
+          # <SUBSCRIPTION_TYPE>_DOWNLOADS="download key"
+          # <SUBSCRIPTION_TYPE>_LOGS="logs key"
+          # (...)
+          source ~/.secrets/dpp-keys
+          ip_addr=$(ip -o -4 addr list eno2 | awk '{print $4}' | cut -d/ -f1)
+
+          robot -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no \
```

Comment:
Is there a reason we do not use https://github.com/Dasharo/open-source-firmware-validation/blob/develop/scripts/run.sh here? We do not want to publish full logs?

But we do not store artifacts anyway?

Please explain it here.

Reply:
We don't want to print full logs because in case of a failure they might leak DPP credentials

Reply:
If running these DTS scripts is very different than running `./run.sh` script on the DTS suite, we may consider adding regression script for DTS as well, so it is easier to run it locally, as well as to have fewer changes in this CI configuration file.

Reply:
But maybe it is not (or does not have to be) and we can simply run the run.sh file?

Reply:
> We don't want to print full logs because in case of a failure they might leak DPP credentials

To the terminal console?

Do we log the DPP variables to the console?

Reply:
We can use https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#masking-a-value-in-a-log to lessen chance that secret information gets logged.
Though I'm not sure if build logs are visible to anyone?